### ISSUE TITLE
startup-script no longer seems to run on host for 1.6

### ISSUE THREAD

**Issue by tonglil** [kubernetes/contrib#2538]
It seems like this daemonset image no longer executes commands on the host w/ K8s 1.6.

https://github.com/kubernetes/contrib/tree/master/startup-script

I have a cluster w/ nodes running 1.6.0 and 1.5.2, and the same daemonset running.

On 1.6.0 (echo is written to /tmp/test inside the container):
```
bash-4.3# ls -la /tmp/
total 12
drwxrwxrwt    1 root     root          4096 Apr  7 21:05 .
drwxr-xr-x    1 root     root          4096 Apr  7 21:05 ..
-rw-r--r--    1 root     root            12 Apr  7 21:05 test
-rw-r--r--    1 root     root             0 Apr  7 21:05 startup-script-v1
```

On 1.5.2 (expected since the commands are executed on the host node):
```
bash-4.3# ls -la /tmp
total 8
drwxrwxrwt    2 root     root          4096 Oct 18 18:58 .
drwxr-xr-x    1 root     root          4096 Apr  7 21:05 ..
```

```yaml
---
kind: DaemonSet
apiVersion: extensions/v1beta1

metadata:
  name: test-ds

spec:
  updateStrategy:
    type: RollingUpdate
    rollingUpdate:
      maxUnavailable: 1

  minReadySeconds: 30

  template:
    metadata:
      labels:
        app: test-ds

    spec:
      hostPID: true
      containers:
        - name: test-ds
          image: gcr.io/google-containers/startup-script:v1
          imagePullPolicy: Always
          resources:
            requests:
              cpu: 10m
            limits:
              cpu: 50m
          securityContext:
            privileged: true
          env:
          # Increment this value to make new DS script apply
          - name: CHECKPOINT_PATH
            value: /tmp/startup-script-v1
          - name: STARTUP_SCRIPT
            value: |
              #!/bin/bash

              set -o nounset
              set -o errexit
              set -o pipefail

              echo "Start"

              echo "hello world" > /tmp/test

              echo "End"
```

**Comment by tonglil** [comment]
It seems like it comes down to `nsenter` no longer executing commands on the host:
```
# 1.6.0
bash-4.3# nsenter -t 1 -m -u -i -n -p -- hostname
test-ds-k9lrg

# 1.5.2
bash-4.3# nsenter -t 1 -m -u -i -n -p -- hostname
gke-tony-test-default-pool-995fe96e-slt7
```

**Comment by roberthbailey** [comment]
@vishh @dchen1107

**Comment by tonglil** [comment]
I can confirm these things:

**1.6.0**

```sh
tony@gke-tony-test-k8s-1-6-34876f29-vs8f /etc/kubernetes/manifests $ docker inspect --format='{{.HostConfig.Privileged}}' k8s_test-ds_test-ds-k9lrg_default_fd96db13-1bd5-11e7-b604-42010a800021_0
true
```

```sh
tony@gke-tony-test-k8s-1-6-34876f29-vs8f /etc/kubernetes/manifests $ sudo ps aux | grep kubelet
root      1358  2.7  2.5 591540 98112 ?        Ssl  Apr07 112:40 /home/kubernetes/bin/kubelet --v=2 --experimental-allocatable-ignore-eviction --allow-privileged=true --babysit-daemons=true --cgroup-root=/ --cloud-provider=gce --cluster-dns=REDACTED --cluster-domain=cluster.local --pod-manifest-path=/etc/kubernetes/manifests --experimental-mounter-path=/home/kubernetes/containerized_mounter/mounter --experimental-check-node-capabilities-before-mount=true --enable-debugging-handlers=true --api-servers=https://REDACTED --anonymous-auth=false --authorization-mode=Webhook --client-ca-file=/etc/srv/kubernetes/pki/ca-certificates.crt --network-plugin-dir=/home/kubernetes/bin --network-plugin=kubenet --node-labels=beta.kubernetes.io/fluentd-ds-ready=true,cloud.google.com/gke-nodepool=k8s-1-6 --eviction-hard=memory.available<100Mi,nodefs.available<10%,nodefs.inodesFree<5% --feature-gates=ExperimentalCriticalPodAnnotation=true
```

**1.5.2**

```sh
tony@gke-tony-test-default-pool-995fe96e-slt7 ~ $ docker inspect --format='{{.HostConfig.Privileged}}' k8s_test-ds.57c33982_test-ds-h9lft_default_fd971452-1bd5-11e7-b604-42010a800021_288639f5
true
```

```sh
tony@gke-tony-test-default-pool-995fe96e-slt7 ~ $ sudo ps aux | grep kubelet
root     30400  1.6  1.5 528256 58688 ?        Ssl  Jan22 1800:37 /home/kubernetes/bin/kubelet --v=2 --allow-privileged=true --babysit-daemons=true --cgroup-root=/ --cloud-provider=gce --cluster-dns=REDACTED  --cluster-domain=cluster.local --config=/etc/kubernetes/manifests --experimental-mounter-path=/home/kubernetes/bin/mounter --experimental-check-node-capabilities-before-mount=true --enable-debugging-handlers=true --api-servers=https://REDACTED  --network-plugin-dir=/home/kubernetes/bin --network-plugin=kubenet --node-labels=cloud.google.com/gke-nodepool=default-pool --eviction-hard=memory.available<100Mi,nodefs.available<10%,nodefs.inodesFree<5%
```

**Comment by tonglil** [comment]
I think the root cause is found here: https://github.com/kubernetes/kubernetes/issues/44041.

**Comment by dchen1107** [comment]
@tonglil Actually you found the root cause of the issue. https://github.com/kubernetes/kubernetes/pull/44097 contains the fix for this, and it is going to be included in 1.6.2 patch release. Thanks!
